util/sst_file_manager_impl.cc

```diff
@@ -130,28 +128,28 @@ void SstFileManagerImpl::OnDeleteFileImpl(const std::string& file_path) {
 SstFileManager* NewSstFileManager(Env* env, std::shared_ptr<Logger> info_log,
                                   std::string trash_dir,
                                   int64_t rate_bytes_per_sec,
-                                  bool delete_existing_trash, Status* status) {
+                                  bool delete_existing_trash, Status* status,
+                                  std::string db_path) {
   SstFileManagerImpl* res =
-      new SstFileManagerImpl(env, info_log, trash_dir, rate_bytes_per_sec);
+      new SstFileManagerImpl(env, info_log, rate_bytes_per_sec);
 
   Status s;
-  if (trash_dir != "") {
-    s = env->CreateDirIfMissing(trash_dir);
+  if (db_path != "") {
+    // Check if there are any files marked as trash in DB directory
+    std::vector<std::string> files_in_db_path;
+    s = env->GetChildren(db_path, &files_in_db_path);
```

Comment:
I don't get how it works when multiple DBs share the same SST File manager.

Reply:
I rethink about it. I think it is OK but maybe we should still call it `trash_dir` rather than `db_path` so that people are not confused. When we move the data to trash dir, we can always add ".trash" suffix and MyRocks can set trash dir as the same as DB dir.

Reply:
@siying, We decided not to use trash directory any more, so we will keep the files in there current location and only add ".trash" to there name.

But your first comment is valid, I should pass a list of directories (multiple DB paths and WAL directories) that we will look for files with ".trash" extension in them to delete them on start up